contracts/FeeCustody.sol

```diff
@@ -0,0 +1,410 @@
+      return IERC20(_asset).balanceOf(address(this)).mul(allocPCT).div(TOTAL_PCT);
+    }
+
+    /**
+     * @notice
+     * Total allocated to RBN lockers across all assets balances
+     * @return total allocated (in USD)
+     */
+    function totalClaimableByRBNLockersInUSD() external view returns (uint256) {
+      uint256 allocPCT = pctAllocationForRBNLockers;
+      return _getSwapQuote(allocPCT);
+    }
+
+    /**
+     * @notice
+     * Total allocated to multisig across all assets balances
+     * @return total allocated (in USD)
+     */
+    function totalClaimableByProtocolInUSD() external view returns (uint256) {
+      uint256 allocPCT = TOTAL_PCT.sub(pctAllocationForRBNLockers);
+      return _getSwapQuote(allocPCT);
+    }
+
+    /**
+     * @notice
+     * Total claimable across all asset balances based on allocation PCT
+     * @param _allocPCT allocation percentage
+     * @return total claimable (in USD)
+     */
+    function _getSwapQuote(uint256 _allocPCT) internal view returns (uint256 claimable) {
+      for(uint i; i < lastAssetIdx; i++){
+        // If we removed asset as part of
+        // protocol revenue distribution, skip
+        if(assetStatus[assets[i]] == -1){
+          continue;
+        }
+
+        IChainlink oracle = IChainlink(oracles[assets[i]]);
+
+        // Approximate claimable by multiplying
+        // current asset balance with current asset price in USD
+        claimable += IERC20(assets[i]).balanceOf(address(this))
+          .mul(oracle.latestAnswer())
+          .mul(_allocPCT)
+          .div(oracle.decimals())
+          .div(TOTAL_PCT);
+      }
+    }
+
+    /**
+     * @notice
+     * Swaps _amountIn of _asset into distributionToken
+     * @param _asset asset to swap from
+     * @param _amountIn amount to swap of asset
+     */
+    function _swap(address _asset, uint256 _amountIn) internal {
+      TransferHelper.safeApprove(asset, address(UNIV3_SWAP_ROUTER), _amountIn);
+
+      address[] memory _intermediaryPath = intermediaryPath[_asset];
+      address[] memory _poolFees = poolFees[_asset];
+
+      // Multiple pool swaps are encoded through bytes called a `path`.
+      // A path is a sequence of token addresses and poolFees that define the pools used in the swaps.
+      // The format for pool encoding is (tokenIn, fee, tokenOut/tokenIn, fee, tokenOut)
+      // where tokenIn/tokenOut parameter is the shared token across the pools.
+
+      bytes pathEncoding;
+      if(_intermediaryPath.length > 0){
+        pathEncoding = abi.encodePacked(_asset, _poolFees[0], _intermediaryPath[0], _poolFees[1], address(distributionToken));
+      }else{
+        pathEncoding = abi.encodePacked(_asset, _poolFees[0], address(distributionToken));
+      }
+
+      ISwapRouter.ExactInputParams memory params =
+          ISwapRouter.ExactInputParams({
+              path: pathEncoding,
+              recipient: msg.sender,
+              deadline: block.timestamp,
```

Comment:
block.timestamp as deadline standard?

Reply:
Unsure if you need to do `block.timestamp + 1` to make this pass

Reply:
Deadline should be passed in via the web3 call / UI and not the smart contract.  I think the default is 20 minutes.